src/entrypoint.sh

```diff
@@ -155,20 +169,40 @@ function git_init() {
     mkdir -p "${HOME}"/.ssh
     ssh-keyscan -t rsa "${source_repo_hostname}" >> "${HOME}"/.ssh/known_hosts
   else
-    info "the source repository is located within GitHub."               
-    if [[ -n "${SOURCE_GH_TOKEN}" ]]; then      
-      unset GITHUB_TOKEN      
-      gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${SOURCE_GH_TOKEN}"        
+    info "the source repository is located within GitHub."
+    # GITHUB_TOKEN is deprecated and can be removed in the future
+    if [[ -n "${SOURCE_GH_TOKEN}" ]] || [[ -n "${GITHUB_TOKEN}" ]]; then
+      ################################
+      if [[ -n "${GITHUB_TOKEN}" ]]; then
+        warn "github_token parameter is deprecated please use source_gh_token."
+        info "setting SOURCE_GH_TOKEN"
+        export SOURCE_GH_TOKEN="${GITHUB_TOKEN}"
+        unset GITHUB_TOKEN
+      fi
+      ###############################
+      if [[ -z "${SOURCE_GH_TOKEN}" ]]; then
+        err "Missing input 'source_gh_token: \${{ secrets.GITHUB_TOKEN }}'.";
+        exit 1;
+      fi
+      info "login to the source git repository"
+      info "source server url: ${source_repo_hostname}"
+      gh auth login --git-protocol "https" --hostname "${source_repo_hostname}" --with-token <<< "${SOURCE_GH_TOKEN}"
+      gh auth status
       if [[ -n "${TARGET_GH_TOKEN}" ]]; then
-        gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${TARGET_GH_TOKEN}"
```

Comment:
actually it is possible that the source is located on another server.

**Remark**
Also possible with a new variable to have the action located in another repository as the target repository. But this includes some further work.

Reply:
This thought came also to my mind. This would be particularly important if you are not using github.com as the host I believe. Something like on-prem hosting would require this if I am not mistaken.